PLUGINS.md

```diff
@@ -1,54 +1,153 @@
+* If the plugin depends on other npm modules, it must contain a `dependencies` section as described [here](https://docs.npmjs.com/files/package.json#dependencies).<br/>The NativeScript CLI will resolve the dependencies during the plugin installation. 
 
-Here is an example. 
+#### Package.json Example
 
-Let you say that you have the following `Android.manifest` file in `platforms\android\` folder in your plugin:
+The following is an example of a `package.json` file for a NativeScript plugin which supports the 1.0.0 version of the iOS runtime and the 1.1.0 version of the Android runtime.
+
+```
+{
+  "name": "myplugin",
+  "version": "0.0.1",
+  "nativescript": {
+    "platforms": {
+      "ios": "1.0.0",
+      "android": "1.1.0"
+    }
+  }
+}
+```
+
+## Install a Plugin
+
+To install a plugin for your project, inside your project, run the following command.
+
+```Shell
+tns plugin add <Plugin>
 ```
+
+### Valid Plugin Sources
+
+You can specify a plugin by name in the npm registry, local path or URL. The following are valid values for the `<Plugin>` attribute.
+
+* A `<Name>` or `<Name>@<Version>` for plugins published in the npm registry.
+* A `<Local Path>` to the directory which contains the plugin files and its `package.json` file.
+* A `<Local Path>` to a `.tar.gz` archive containing a directory with the plugin and its `package.json` file.
+* A `<URL>` which resolves to a `.tar.gz` archive containing a directory with the plugin and its `package.json` file.
+* A `<git Remote URL>` which resolves to a `.tar.gz` archive containing a directory with the plugin and its `package.json` file.
+
+### Installation Specifics
+
+The installation of a NativeScript plugin mimics the installation of an npm module.
+
+The NativeScript CLI takes the plugin and copies it to the `app\tns_modules` directory inside your project. During this process, the NativeScript CLI resolves any dependencies described in the plugin `package.json` file and adds the plugin to the project `package.json` file in the project root.
+
+Next, the NativeScript CLI runs `$ tns prepare` for all platforms configured for the project. During this operation, the CLI copies the plugin to the `tns_modules` subdirectories in the `platforms\android` and `platform\ios` directories in your project. If your plugin contains platform-specific `JS` files, the CLI copies them to the respective platform subdirectory and renames them by removing the platform modifier. Finally, the CLI merges the plugin `AndroidManifest.xml` and `Info.plist` files with `platforms\android\AndroidManifest.xml` and `platforms\ios\Info.plist` in your project. 
```

Comment:
`Next, the NativeScript CLI runs `$ tns prepare`` - Actually cli runs `tns plugin prepare` which is internal "command".